test/PreAMMBatcher.spec.ts

```diff
@@ -0,0 +1,105 @@
+import {use, expect} from 'chai';
+import {Contract, utils} from 'ethers';
+import {deployContract, deployMockContract, MockProvider, solidity} from 'ethereum-waffle';
+import PreAMMBatcher from '../build/PreAMMBatcher.json';
+import UniswapV2Pair from '../node_modules/@uniswap/v2-core/build/UniswapV2Pair.json';
+import UniswapV2Factory from '../node_modules/@uniswap/v2-core/build/UniswapV2Factory.json';
+
+import ERC20 from '../node_modules/@openzeppelin/contracts/build/contracts/ERC20Mintable.json';
+import {Order, DOMAIN_SEPARATOR} from '../src/js/orders.spec';
+import BN from 'bn.js';
+
+use(solidity);
+
+describe('PreAMMBatcher', () => {
+  const [walletDeployer, walletTrader1, walletTrader2] = new MockProvider().getWallets();
+  let batcher: Contract;
+  let token0: Contract;
+  let token1: Contract;
+  let uniswapPair: Contract;
+  let uniswapFactory: Contract;
+  let uniswapPairAddress: string;
+  const initial_uniswap_funding_token0 = utils.parseEther('10');
+  const initial_uniswap_funding_token1 = utils.parseEther('10');
+
+  const mockRelevantTokenDataForPreBatchTrade = async (sellOrder1: Order, sellOrder2: Order) => {
+    // todo: mock actual right information
+    // mock un-relevant data to make preBatchTrade call pass
+    await token0.mock.transferFrom.returns(true);
+    await token1.mock.transferFrom.returns(true);
+    await token0.mock.transfer.returns(true);
+    await token1.mock.transfer.returns(true);
+    await token0.mock.balanceOf.withArgs(sellOrder1.wallet.address).returns(sellOrder1.sellAmount);
+    await token1.mock.balanceOf.withArgs(sellOrder2.wallet.address).returns(sellOrder2.sellAmount);
+    await token0.mock.balanceOf.withArgs(batcher.address).returns('1');
+    await token1.mock.balanceOf.withArgs(batcher.address).returns('1');
+    await token0.mock.balanceOf.withArgs(uniswapPair.address)
+      .returns(new BN(initial_uniswap_funding_token0.toString()).add(sellOrder1.sellAmount).toString());
+    await token1.mock.balanceOf.withArgs(uniswapPair.address).returns(initial_uniswap_funding_token1);
+  };
+
+  beforeEach(async () => {
```

Comment:
Maybe include a comment here like 
```suggestion
  beforeEach(async () => {
    \\ deploy all relevant contracts and setup the uniswap pool
```

Reply:
Sorry, I think my slashes were backwards!